gsy_myco_sdk/matching_algorithms/preferred_partners_algorithm.py

```diff
@@ -206,7 +206,7 @@ def _match_one_bid_one_offer(
                 recommendation = BidOfferMatch(
                     market_id=market_id,
                     time_slot=time_slot,
-                    bid=bid, offer=offer,
+                    bid=deepcopy(bid), offer=deepcopy(offer),
```

Comment:
Why don't we put this logic inside `BidOfferMatch`? So that we won't have to worry about it for other cases too.

Reply:
I would also vote for that. Thanks!

Reply:
Thanks a lot, seems like a good idea https://github.com/gridsingularity/gsy-framework/pull/340